pyiron_continuum/schroedinger/schroedinger.py

```diff
@@ -0,0 +1,268 @@
+    )
+
+
+# TODO: Convert to pyiron units
+HBAR = 1  # EV_TO_U_ANGSTROMSQ_PER_SSQ * physical_constants['Planck constant in eV/Hz'][0] / (2 * np.pi)
+M_EL = 1  # physical_constants['electron mass in u'][0]
+KB = physical_constants['Boltzmann constant in eV/K'][0]
+
+__author__ = "Liam Huber"
+__copyright__ = (
+    "Copyright 2021, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "0.1"
+__maintainer__ = "Liam Huber"
+__email__ = "huber@mpie.de"
+__status__ = "development"
+__date__ = "Sep 8, 2021"
+
+
+class _TISEInput(DataContainer):
+    def __init__(self, init=None, table_name='input', lazy=False):
+        super().__init__(init=init, table_name=table_name, lazy=lazy)
+        self._mesh = None
+        self._potential = None
+        self.n_states = 1
+        self.mass = M_EL
+
+    @property
+    def mesh(self) -> RectMesh:
+        return self._mesh
+
+    @mesh.setter
+    def mesh(self, new_mesh: RectMesh):
+        if not isinstance(new_mesh, RectMesh):
+            raise TypeError(f'Meshes must be of type {RectMesh} but got type {type(new_mesh)}')
+        new_mesh.simplify_1d = False
+        self._mesh = new_mesh
+
+    @property
+    def potential(self):
+        return self._potential
+
+    @potential.setter
+    def potential(self, new_potential):
+        self._potential = new_potential
+
+
+class _TISEOutput(DataContainer):
+    def __init__(self, init=None, table_name='input', lazy=False):
+        super().__init__(init=init, table_name=table_name, lazy=lazy)
+        self.energy = None
+        self.psi = None
+        self._rho = None
+
+    @property
+    def _space_axes(self):
+        return tuple(n for n in range(1, len(self.psi.shape)))
+
+    def _broadcast_vector_to_space(self, vector):
+        return vector.reshape(-1, *(1,)*len(self._space_axes))
+
+    def _normalize_per_state(self, states):
+        return states / self._broadcast_vector_to_space(np.linalg.norm(states, axis=self._space_axes))
+
+    def _weight_states(self, states, weights):
+        return states * self._broadcast_vector_to_space(weights)
+
+    @property
+    def rho(self):
+        if self._rho is None and self.psi is not None:
+            self._rho = self._normalize_per_state(self.psi ** 2)
+        return self._rho
+
+    def get_boltzmann_occupation(self, temperature):
+        if self.energy is not None:
+            return 1. / (np.exp(self.energy / (KB * temperature)) + 1)
+
+    def get_boltzmann_psi(self, temperature):
+        if self.psi is not None:
+            weighted_psi = self._weight_states(self.psi, self.get_boltzmann_occupation(temperature))
+            psi_tot = np.sum(weighted_psi, axis=0)
+            return psi_tot / np.linalg.norm(psi_tot)
+
+    def get_boltzmann_rho(self, temperature):
+        if self.psi is not None:
+            weighted_psi = self._weight_states(self.psi, self.get_boltzmann_occupation(temperature))
```

Comment:
I tried this for my work case of a Murnaghan-type potential, and the resulting weighted PD was a proper smooth (skewed) distribution. However, with increasing temperatures, the curve had a very different variance when compared to a classical Maxwell-Boltzmann solution, where we would want to see a rather close agreement (wrt our project).

With my very limited knowledge of Quantum Mechanics and a month old experience with solving TISEs, I still believe that since we are interested in the 'weighted' PD, we must weight the PD of each eigenstate before summing them, rather than weighting the eigenstates and then summing the square of the weighted eigenstates, as done here and the next line. I also see that the weighted eigenstate is not orthogonal to the other eigenstates (I don't really know if it should be in this case). 

If I weight the PD of each eigenstate before summing them, I get almost identical distributions for the quantum and classical solutions at higher temperatures. This is why I believe it warrants some more thought and discussion.

Reply:
This is your long comment, but it's the next one that's actually the critical one ;)